### ISSUE TITLE
[Bug]: Dropdown does not have focus indicators in Strict Mode

### ISSUE THREAD

**Issue by layershifter** [microsoft/fluentui#31294]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://stackblitz.com/edit/hqj6td?file=src%2Fexample.tsx

## Actual Behavior
- Navigate to Dropdown
- Press <kbd>Enter</kbd>
- Try to navigate using <kbd>ArrowDown</kbd> or <kbd>ArrowUp</kbd>
- No focus indicators 💥 

## Expected Behavior
- Focus indicators are present

### Logs

```shell
N/A
```

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
